### DISCUSSION TITLE
Get katexEqnNo to reference equations in TiddlyWiki

### DISCUSSION THREAD

**Discussion by Diasaur** [KaTeX/KaTeX#3910] [Q&A]
I use KaTeX with TidllyWIki (https://kookma-katex.tiddlyhost.com/) and some comment considering older TiddlyWiki versions has brought to my attention, that KaTeX uses the css counter "katexEqnNo" to count through equations. This had me thinking: There surely is a way to save the current counter number as in:
`myLabel = katexEqnNo `
Then later one could reference said equation as equation `(myLabel)`.

I have tried several methods of getting the value of katexEqnNo but my HTML and CSS are very rusty. This seemed like it should be doable, so I'm grateful for any ideas as to how to obtain the current katexEqnNo for a certain equation.

**Comment by mbourne** [comment]
@Diasaur 

You can't get the equation numbers using javascript since CSS counters don't even belong in the DOM.

So here's one way you might go about it. We find all the tags and their numbers, and keep track of which number goes with which equation. This may or may not do what you need.

    document.addEventListener("DOMContentLoaded", function() {
        renderMathInElement(document.body, {
          // Standard options::
          delimiters: [
              {left: '$$', right: '$$', display: true},
              {left: '$', right: '$', display: false},
              {left: '\\(', right: '\\)', display: false},
              {left: '\\[', right: '\\]', display: true}
          ],
          throwOnError : false
        });

        // The math is now processed, so we can find all the tags:
        var tags = document.querySelectorAll('span.tag');
        
        // Loop through the tags and get the (zero-indexed) tag numbers:
        // Also, get the original LaTeX for the equation to keep track of which tag number goes with which equation
        tags.forEach(function(tag, i) {
              console.log("Tag number "+(i+1)+": "+ tag.closest('span.katex').querySelector('annotation').textContent)
        });	
    });

This will only work if you're using simple counters. Hope it's of some use.
Regards
Murray

**Reply by Diasaur** [reply]
Sorry for the delay over the holidays and thank you so much for your answer!

Sadly my CSS (or probably mostly TiddlyWiki...) skills are not high enough to implement this at the moment, but maybe I'll ask the folks on the TiddlyWiki side and they can help me on implementation :)
I'll leave this open for now and will come back once I found a solution with or without this that works for my use case so that people with the same problem will have a reference here.

Regards
Darius
